src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaFileProvider.cs

```diff
@@ -28,16 +30,22 @@ public MediaFileProvider(IMediaFileStore mediaStore, IOptions<ImageSharpMiddlewa
         public IDictionary<string, string> Settings { get; set; } = new Dictionary<string, string>();
 
         /// <inheritdoc/>
-        public bool IsValidRequest(HttpContext context) => _formatHelper.GetExtension(context.Request.GetDisplayUrl()) != null;
+        public bool IsValidRequest(HttpContext context) => _formatUtilities.GetExtensionFromUri(context.Request.GetDisplayUrl()) != null;
 
         /// <inheritdoc/>
-        public IImageResolver Get(HttpContext context)
+        public async Task<IImageResolver> GetAsync(HttpContext context)
         {
             // Path has already been correctly parsed before here.
-
             var filePath = _mediaStore.MapPublicUrlToPath(context.Request.PathBase + context.Request.Path.Value);
 
-            return new MediaFileResolver(_mediaStore, filePath);
+            // Check to see if the file exists.
+            var file = await _mediaStore.GetFileInfoAsync(filePath);
+            if (file == null)
+            {
+                return null;
+            }
+            var metadata = new ImageMetaData(file.LastModifiedUtc);
+            return await Task.FromResult<IImageResolver>(new MediaFileResolver(_mediaStore, filePath, metadata));
```

Comment:
```suggestion
            return new MediaFileResolver(_mediaStore, filePath, metadata);
```

Reply:
hi @sebastienros ,what's the difference here?

Reply:
You don't need to await a completed task. And a constant is already completed. Because you have another await in the code you can just return the value.

Reply:
thanks @sebastienros ,great explanation and I read some [ms doc](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/async-return-types) .